### PR TITLE
Bump to depend on Airbnb 17.0.1

### PR DESCRIPTION
**Pull request by justin808**
Ignoring max-len for strings and template literals makes it too easy
to include long lines in the project.

However, if one is using prettier settings with eslint, max-len gets
ingored anyway.

---
This change is [Reviewable](https://reviewable.io/reviews/shakacode/style-guide-javascript/27)

### CODE REVIEW

File: packages/eslint-config-shakacode/rules/javascript.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ rules:
 
   # http://eslint.org/docs/rules/max-len
   # https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb-base/rules/style.js#L128
-  # Almost identical to airbnb's version, but adds ignore of `import` statements
+  # Almost identical to airbnb's version, but adds ignore of `import` statements and sets ignoreStrings: false
```
Comment (alex35mil) [inline packages/eslint-config-shakacode/rules/javascript.yml]: I think airbnb version is correct?

File: packages/eslint-config-shakacode/package.json [hunk 2]
```diff
@@ -22,16 +22,13 @@
     "release:major": "scripts/release major"
   },
   "peerDependencies": {
-    "eslint": "^4.9.0"
+    "eslint": "^4.19.1",
+    "eslint-plugin-import": "^2.12.0",
+    "eslint-plugin-jsx-a11y": "^6.0.3",
+    "eslint-plugin-react": "^7.9.1"
```
Comment (alex35mil) [inline packages/eslint-config-shakacode/package.json]: React deps are optional.

Comment (justin808) [reply packages/eslint-config-shakacode/package.json]: I copied these from [Airbnb's one](https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb/package.json#L76).

File: packages/eslint-config-shakacode/rules/javascript.yml (hunk 1 above)
Comment (justin808) [reply packages/eslint-config-shakacode/rules/javascript.yml]: This is Airbnb: https://github.com/airbnb/javascript/blob/17e0454672f784772c5f5ab052523a93229d9adf/packages/eslint-config-airbnb-base/rules/style.js#L195

If you prefer, I'll remove these for now.

```
    // specify the maximum length of a line in your program
    // https://eslint.org/docs/rules/max-len
    'max-len': ['error', 100, 2, {
      ignoreUrls: true,
      ignoreComments: false,
      ignoreRegExpLiterals: true,
      ignoreStrings: true,
      ignoreTemplateLiterals: true,
    }],
```

**Comment by justin808** [comment]
@alexfedoseev I'm merging. We can do any review comments in a follow-on PR.

**Pull request merged**